flow/connectors/postgres/qrep.go

```diff
@@ -186,6 +188,19 @@ func (c *PostgresConnector) getNumRowsPartitions(
 	}
 
 	if err := rows.Err(); err != nil {
+		var pgErr *pgconn.PgError
+		if config.WatermarkColumn == "ctid" && errors.As(err, &pgErr) &&
+			pgErr.Code == "XX000" && pgErr.Message == "transparent decompression only supports tableoid system column" {
```

Comment:
it looks like it can also be "42P10"
also can use `pgerrcode.InternalError` instead

Reply:
since we have to do a message check anyways might as well drop code check in that case